src/components/overview/Overview.jsx

```diff
@@ -33,14 +32,43 @@ const InvisDiv = styled.div`
   margin: 5px;
 `;
 
-function Overview({ productId, ratings }) {
-  const [currentProduct, setCurrentProduct] = useState(exampleData.productData);
-  const [styles, setStyles] = useState(exampleData.stylesData);
-  const [currentStyle, setCurrentStyle] = useState(exampleData.stylesData[0]);
+function Overview({ id }) {
+  const [currentProduct, setCurrentProduct] = useState({});
+  const [styles, setStyles] = useState([]);
+  const [currentStyle, setCurrentStyle] = useState(undefined);
   const [currentPhoto, setCurrentPhoto] = useState('');
 
-  const handleStyleOnClick = (id) => {
-    setCurrentStyle(styles.find((item) => item.style_id === id));
+  const handleImage = () => {
+    const promises = styles.map((style) => (
+      new Promise((resolve, reject) => {
+        const img = new Image();
+        img.src = style.photos[0].url;
+        img.onload = resolve();
+        img.onerror = reject();
+      })
+    ));
+    Promise.all(promises);
+  };
+
+  useEffect(() => {
+    const promises = [];
+    promises.push(
+      getProduct(id)
+        .then((res) => {
+          setCurrentProduct(res);
+          setStyles(res.styles);
+          setCurrentStyle(res.styles.find((style) => style['default?'] === true) || res.styles[0]);
+        })
+        .catch((err) => Error('Error in Overview getProduct', err)),
+    );
+    Promise.all(promises)
+      .then(() => {
+        handleImage();
+      });
+  }, []);
```

Comment:
I think `id` should be added to this `useEffect`'s array so that the widget can update to load another product if the user navigates to a different product.